### PR TITLE
[docs] Link docs from component demos on Toolpad Core landing page

### PR DESCRIPTION
**Pull request by prakhargupta1**
Preview: https://deploy-preview-4013--mui-toolpad-docs.netlify.app/toolpad/

### CODE REVIEW

**Comment by bharatkashyap** [comment]
I think we could go with this kind of an overlay on the code for the links: much more unmissable and attractive:
Screenshot 2024-08-29 at 1 47 59 PM

**Comment by prakhargupta1** [comment]
I don't think it will bring much difference.

File: docs/src/components/landing/ToolpadFeaturesSwitcher.tsx [hunk 1]
```diff
@@ -57,6 +61,12 @@ function ComponentItem({
             {description}
           </Typography>
         )}
+        {link && (
+          <Link href={link.toString()} underline="hover" variant="body2" target="_blank">
+            View docs
```
Comment (Janpot) [inline docs/src/components/landing/ToolpadFeaturesSwitcher.tsx:66]:
```suggestion
            View the docs
```

to mimic X https://mui.com/x/

**Review by Janpot — COMMENTED** [review]
> I think we could go with this kind of an overlay on the code for the links: much more unmissable and attractive:

Personally I find a link more intuitive than a button as a way to get to a page in our documentation. This follows the [X landing page](https://mui.com/x/), it feels better to me than to core landing page.

File: docs/src/components/landing/ToolpadFeaturesSwitcher.tsx [hunk 2]
```diff
@@ -3,7 +3,9 @@ import dynamic from 'next/dynamic';
 import Box from '@mui/material/Box';
 import Stack from '@mui/material/Stack';
 import Typography from '@mui/material/Typography';
+import Link from '@mui/material/Link';
```
Comment (Janpot) [inline docs/src/components/landing/ToolpadFeaturesSwitcher.tsx:6]: This page lives in the same Next.js app as the docs, so it's an internal link. We could avoid a full navigation if we use the Next.js router. Is it possible to instead use
```suggestion
import { Link } from '@mui/docs/Link';
```

File: docs/src/components/landing/ToolpadFeaturesSwitcher.tsx [hunk 3]
```diff
@@ -57,6 +61,12 @@ function ComponentItem({
             {description}
           </Typography>
         )}
+        {link && (
+          <Link href={link.toString()} underline="hover" variant="body2" target="_blank">
```
Comment (Janpot) [inline docs/src/components/landing/ToolpadFeaturesSwitcher.tsx:65]:
```suggestion
          <Link href={link.toString()} underline="hover" variant="body2">
```

Comment (prakhargupta1) [reply docs/src/components/landing/ToolpadFeaturesSwitcher.tsx:65]: Got it, that's how it's in MUI X.

**Review by Janpot — APPROVED** [review]

**Pull request merged**